### ISSUE TITLE
oxAuth reloads custom scripts (file method)

### ISSUE THREAD

**Issue by sandovalezequiel** [GluuFederation/oxAuth#1460]
Build: gluu-server-4.2.1-403

## Description
When I enable a script with file method, oxAuth reload that script each 30 secs. Scripts that are loaded in oxTrust (DB method) are not re-loaded, just the file ones.

## Steps To Reproduce
1. Check oxauth_script.log file
2. Enable a script with file method

## Expected
Script is loaded when is enabled or each time that file has some change.

## Actual behavior
Script enabled with file method is reloaded each 30 secs until oxAuth is stopped because a `Failed to load python file
org.python.core.PyException: RuntimeError: maximum recursion depth exceeded` exception.

**Comment by jgomer2001** [comment]
I think this [commit](https://github.com/GluuFederation/oxCore/commit/6b755a1b18d1127bb6a57767f33eead6b1d7fd13) has a problem: Even if `hasChanged` is equal to  `false` at this [point](https://github.com/GluuFederation/oxCore/blob/version_4.2.1/core-script/src/main/java/org/gluu/service/custom/script/CustomScriptManager.java#L234), script reload will take place if location type  is FILE.

Can you check @syntrydy ?

**Comment by jgomer2001** [comment]
btw these 2 blocks of code are suspiciously similar...
- https://github.com/GluuFederation/oxCore/blob/version_4.2.1/core-script/src/main/java/org/gluu/service/custom/script/CustomScriptManager.java#L220-L224
- https://github.com/GluuFederation/oxCore/blob/version_4.2.1/core-script/src/main/java/org/gluu/service/custom/script/CustomScriptManager.java#L259-L262

**Comment by moabu** [comment]
Issue can be seen below

```
{"message":"2020-10-22 16:45:00,149 INFO  [oxAuthScheduler_Worker-5] - Resource Owner Password:Initialization success"}
{"message":"2020-10-22 16:45:19,750 INFO  [oxAuthScheduler_Worker-4] - NAP Session Management:Initializing ..."}
{"message":"2020-10-22 16:45:19,754 INFO  [oxAuthScheduler_Worker-4] - no_session:NAP Introspection:Destroying ..."}
{"message":"2020-10-22 16:45:20,692 INFO  [oxAuthScheduler_Worker-4] - no_session:MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:45:20,699 INFO  [oxAuthScheduler_Worker-4] - no_session:NAP Introspection:Initializing ..."}
{"message":"2020-10-22 16:45:24,573 INFO  [oxAuthScheduler_Worker-4] - no_session:MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:45:24,580 INFO  [oxAuthScheduler_Worker-4] - no_session:Id First Passport:Initialization success"}
{"message":"2020-10-22 16:45:28,367 INFO  [oxAuthScheduler_Worker-4] - no_session:Admin Passport:Initialization success"}
{"message":"2020-10-22 16:45:28,372 INFO  [oxAuthScheduler_Worker-4] - Resource Owner Password:Destroying ..."}
{"message":"2020-10-22 16:45:29,490 INFO  [oxAuthScheduler_Worker-4] - MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:45:29,496 INFO  [oxAuthScheduler_Worker-4] - Resource Owner Password:Initialization success"}
{"message":"2020-10-22 16:45:49,731 INFO  [oxAuthScheduler_Worker-3] - NAP Session Management:Initializing ..."}
{"message":"2020-10-22 16:45:49,737 INFO  [oxAuthScheduler_Worker-3] - no_session:NAP Introspection:Destroying ..."}
{"message":"2020-10-22 16:45:50,664 INFO  [oxAuthScheduler_Worker-3] - no_session:MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:45:50,670 INFO  [oxAuthScheduler_Worker-3] - no_session:NAP Introspection:Initializing ..."}
{"message":"2020-10-22 16:45:55,553 INFO  [oxAuthScheduler_Worker-3] - no_session:MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:45:55,560 INFO  [oxAuthScheduler_Worker-3] - no_session:Id First Passport:Initialization success"}
{"message":"2020-10-22 16:45:59,648 INFO  [oxAuthScheduler_Worker-3] - no_session:Admin Passport:Initialization success"}
{"message":"2020-10-22 16:45:59,656 INFO  [oxAuthScheduler_Worker-3] - Resource Owner Password:Destroying ..."}
{"message":"2020-10-22 16:46:00,561 INFO  [oxAuthScheduler_Worker-3] - MonolithAuthentication:Checking monolith connection"}
{"message":"2020-10-22 16:46:00,591 INFO  [oxAuthScheduler_Worker-3] - Resource Owner Password:Initialization success"}
```

This eventually causes oxAuth to crash.

**Comment by yuriyz** [comment]
The root cause is in @syntrydy 's commit, so we need his opinion.

https://github.com/GluuFederation/oxCore/commit/6b755a1b18d1127bb6a57767f33eead6b1d7fd13

**Comment by syntrydy** [comment]
Fixed in 4.2.2 & master

**Comment by moabu** [comment]
commit that was addressed by @syntrydy https://github.com/GluuFederation/oxCore/commit/263cf44be7e9ec2053082564a85bc9817002809e
